clients/src/main/java/org/apache/kafka/common/config/ConfigDef.java

```diff
@@ -267,11 +423,23 @@ else if (value instanceof String)
         HIGH, MEDIUM, LOW
     }
 
+    public enum Width {
+        NONE, SHORT, MEDIUM, LONG
+    }
+
+    /**
+     *
+     */
+    public interface Recommender {
+        Set<Object> validValues(String name, String parentName, Object parentValue);
```

Comment:
Is only including the parent configuration actually sufficient? Any reason not to include the full context of the parsed config values? In particular, I'm wondering about 2 situations. First, to get valid values for some systems, you'll always need something like connection URL. This one config will have to be the parent of every other config, and based on this interface it looks like it'll have to be the direct parent (e.g., not a grandparent). Second, What about cases where some relevant info might be split across multiple fields. For example, what if username, password, and connection URL are separate? Often you can include username + password in the URL, but it may not always be possible and in some cases you might want to keep the password separate so it can be protected. In this case you would need something like multiple parents rather than just one.

I'd also suggest renaming `validValues` to something like `recommendedValues` to be consistent with the name of the interface. "valid" is used to mean something different in a different interface, so it'd be confusing to reuse it.